app/v2/notifications/post_notifications.py

```diff
@@ -74,8 +74,9 @@ def post_precompiled_letter_notification():
 
     form = validate(request.get_json(), post_precompiled_letter_request)
 
-    # Check permission to send letters
-    check_service_has_permission(LETTER_TYPE, authenticated_service.permissions)
+    if not authenticated_service.has_permissions(LETTER_TYPE):
+        raise BadRequestError(message="Service is not allowed to send {}".format(
+            get_public_notify_type_text(LETTER_TYPE, plural=True)))
```

Comment:
Why .format instead of an f-string?

Reply:
That code is copy-pasted from another file written before my time here, so I don't have a good answer for that.

Reply:
I'm going to resolve these comments because .format and f string do the same thing and there's no benefit performance-wise. Changing to f-string would make string longer because payload in a function.